### PR TITLE
Encode lossless Modular extra channels in streaming mode.

### PR DESCRIPTION
**Pull request by veluca93**

### CODE REVIEW

File: lib/jxl/enc_frame.cc [hunk 1]
```diff
@@ -1870,7 +1882,7 @@ Status EncodeFrameStreaming(const CompressParams& cparams,
   ComputePermutationForStreaming(frame_data.xsize, frame_data.ysize, num_passes,
                                  permutation, dc_group_order);
   enc_state.shared.num_histograms = dc_group_order.size();
-  // This is only valid in VarDCT mode, otherwise there can be group shift.
+  // This is only valid with a group shift of 1.
```
Comment (szabadka) [inline lib/jxl/enc_frame.cc:1885]: Shouldn't this condition be added to CanDoStreamingEncoding() ?

File: lib/jxl/enc_modular.cc [hunk 2]
```diff
@@ -373,29 +374,31 @@ bool do_transform(Image& image, const Transform& tr,
 Status ModularFrameEncoder::ComputeEncodingData(
     const FrameHeader& frame_header, const ImageMetadata& metadata,
     Image3F* JXL_RESTRICT color, const std::vector<ImageF>& extra_channels,
-    PassesEncoderState* JXL_RESTRICT enc_state, const JxlCmsInterface& cms,
-    ThreadPool* pool, AuxOut* aux_out, bool do_color) {
+    const Rect& group_rect, const FrameDimensions& patch_dim,
+    const Rect& frame_area_rect, PassesEncoderState* JXL_RESTRICT enc_state,
+    const JxlCmsInterface& cms, ThreadPool* pool, AuxOut* aux_out,
+    bool do_color, bool groupwise) {
```
Comment (szabadka) [inline lib/jxl/enc_modular.cc:380]: Why do we need groupwise parameter? It seems to be redundant with enc_state->streaming_mode

File: lib/jxl/enc_modular.cc [hunk 3]
```diff
@@ -1104,6 +1108,24 @@ void ModularFrameEncoder::ClearStreamData(const ModularStreamId& stream) {
...
+void ModularFrameEncoder::ClearModularStreamData() {
+  for (const auto& group : stream_params_) {
+    ClearStreamData(group.id);
+  }
+  stream_params_.clear();
+}
+
+size_t ModularFrameEncoder::ComputeStreamingAbsoluteAcGroupId(
+    size_t dc_group_id, size_t ac_group_id,
+    const FrameDimensions& dc_dims) const {
```
Comment (szabadka) [inline lib/jxl/enc_modular.cc:1120]: patch_dim? (to be consistent with ComputeEncodingData())

File: lib/jxl/enc_modular.cc (hunk 2 above)
Comment (veluca93) [reply lib/jxl/enc_modular.cc]: good question...

**Review by szabadka — APPROVED** [review]

**Pull request merged**